src/globus_sdk/experimental/globus_app/globus_app.py

```diff
@@ -37,35 +42,17 @@
 from .errors import IdentityMismatchError, TokenValidationError
 
 if sys.version_info < (3, 8):
-    from typing_extensions import Protocol
+    from typing_extensions import Protocol, runtime_checkable
 else:
-    from typing import Protocol
-
+    from typing import Protocol, runtime_checkable
 
-def _default_filename(app_name: str, environment: str) -> str:
-    r"""
-    construct the filename for the default JSONTokenStorage to use
 
-    on Windows, this is typically
-        ~\AppData\Local\globus\app\{app_name}/tokens.json
-
-    on Linux and macOS, we use
-        ~/.globus/app/{app_name}/tokens.json
-    """
-    environment_prefix = f"{environment}-"
-    if environment == "production":
-        environment_prefix = ""
-    filename = f"{environment_prefix}tokens.json"
-
-    if sys.platform == "win32":
-        # try to get the app data dir, preferring the local appdata
-        datadir = os.getenv("LOCALAPPDATA", os.getenv("APPDATA"))
-        if not datadir:
-            home = os.path.expanduser("~")
-            datadir = os.path.join(home, "AppData", "Local")
-        return os.path.join(datadir, "globus", "app", app_name, filename)
-    else:
-        return os.path.expanduser(f"~/.globus/app/{app_name}/{filename}")
+@runtime_checkable
+class TokenStorageProvidable(Protocol):
```

Comment:
"Providable" is an unusual relationship to declare. 
It looks to me like this is a "provider", "factory", "generator", etc. Is that understanding correct?

A naming scheme I've seen used for protocols in places like typeshed is `SupportsX`. So maybe this could be `SupportsAppTokenStorage`?

I'm not exactly against the current name, but I don't understand it. This seems to me like a case we should clarify, and possibly rename of we decide that there's a better choice.

Reply:
Yeah, I'm not entirely sure why I went with "Providable" in the end. It is a Provider but in Protocol form so I froze.

I added a similar thing to LoginFlowManagers in pr-1018 & called that one a `LoginFlowManagerProvider`. Are you happy with that?